provider/offline_test.go

```diff
@@ -200,67 +205,74 @@ func TestOfflineStores(t *testing.T) {
 	if err != nil {
 		panic(err)
 	}
+
 	for _, testItem := range testList {
-		if testing.Short() && testItem.integrationTest {
-			t.Logf("Skipping %s, because it is an integration test", testItem.t)
-			continue
-		}
-		var connections_start, connections_end int
-		if testItem.t == PostgresOffline {
-			err = db.QueryRow("SELECT Count(*) FROM pg_stat_activity WHERE pid!=pg_backend_pid()").Scan(&connections_start)
-			if err != nil {
-				panic(err)
-			}
+		// for running go routines inside for loops in go, the iterated value needs to be redeclared
+		// this prevents the earlier go routines from referencing values in a later iteration of the for loop
+		testItemConst := testItem
+		t.Run(string(testItemConst.t), func(t *testing.T) {
+			t.Parallel()
+			testWithProvider(t, testItemConst, testFns, testSQLFns, db)
+		})
+	}
+}
+
+func testWithProvider(t *testing.T, testItem testMember, testFns map[string]func(*testing.T, OfflineStore), testSQLFns map[string]func(*testing.T, OfflineStore), db *sql.DB) {
```

Comment:
Does this have to be its own function?

Reply:
no, but I separated it  out so the for loop inside it that iterates over all the test types doesn't get nestled too deep